protostar/starknet/abi.py

```diff
@@ -28,10 +30,13 @@ def get_function_parameters(contract_abi: AbiType, name: str) -> Dict[str, Cairo
 
 
 def find_abi_item(contract_abi: AbiType, name: str) -> Dict:
-    for item in contract_abi:
-        if item["name"] == name:
-            return item
-    raise AbiItemNotFoundException(f"Couldn't find '{name}' ABI")
+    try:
+        for item in contract_abi:
+            if item["name"] == name:
+                return item
+        raise AbiItemNotFoundException(f"Couldn't find '{name}' ABI")
+    except TypeError as ex:
```

Comment:
Could you please explain where possibly a `TypeError` could raise here?

Reply:
If I remember correctly, `item["name"] == name` can raise `TypeError`. The error is thrown in `test_error_when_custom_abi_is_invalid`.

Reply:
I think we shouldn't catch a `TypeError` like this, I'd rather validate the abi JSON in the place where it is parsed `compiled_contract_reader.py::load_abi`. It seems that a valid abi looks like this `[{'inputs': [{...}, ...]}, {'outputs': [{...}, ...]}, {...}, ...]` while an invalid one like this `{'abi': [{'inputs': [{...}, ...]}, {'outputs': [{...}, ...]}, {...}, ...]}`.

I think we should either immediately throw or maybe fallback like this:
```
if isinstance(abi, 'dict'):
  abi = abi.get("abi")
```
What you're doing right now is you're validating the abi JSON a couple of levels deeper into the implementation, which is bad because you're handling an invalid input only for one case.

Reply:
[use load_abi](https://github.com/software-mansion/protostar/pull/1373/commits/6cec71984b61c830903a01eeaae5b6372e761186)